pkg/middleware/response_writer.go

```diff
@@ -0,0 +1,68 @@
+package middleware
+
+import (
+	"bufio"
+	"errors"
+	"net"
+	"net/http"
+
+	middlewareapi "github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/middleware"
+)
+
+type responseWriter struct {
+	http.ResponseWriter
+
+	status int
+	size   int
+}
+
+func NewResponseWriter(rw http.ResponseWriter) middlewareapi.ResponseWriter {
```

Comment:
Ahh the other was in the API package, my temptation is to just make this all internal, who outside of the request logger needs this?